binder/src/main/java/io/grpc/binder/internal/BinderClientTransportFactory.java

```diff
@@ -191,5 +194,28 @@ public Builder setBinderDecorator(OneWayBinderProxy.Decorator binderDecorator) {
       this.binderDecorator = checkNotNull(binderDecorator, "binderDecorator");
       return this;
     }
+
+    /**
+     * Limits how long it can take to for a new transport to become ready after being started.
+     *
+     * <p>This process currently includes:
+     * <ul>
+     * <li>Creating an Android binding.
+     * <li>Waiting for Android to create the server process.
+     * <li>Waiting for the remote Service to be created and handle onBind().
+     * <li>Exchanging handshake transactions according to the wire protocol.
+     * <li>Evaluating a {@link SecurityPolicy} on both sides.
+     * </ul>
+     *
+     * <p>This setting doesn't change the need for deadlines at the call level. It merely ensures
+     * that gRPC features like load balancing and wait-for-ready work as expected despite certain
```

Comment:
This shouldn't impact wait-for-ready. It would impact "non-wait-for-ready" RPCs.

Reply:
Agreed that when the target resolves to a single unresponsive server, the outcome of a withWaitForReady() call is the same with or without this PR -- it hangs forever. But in this sentence I want to talk about the "wait for ready" feature/distinction in general and I think it's clearer to refer to it in the more concise positive sense the same way we do in the docs (https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md) even though withoutWaitForReady() is the default. But LMK if you feel strongly.

Reply:
I don't think any of us would read it that way. The wait-for-ready feature is only involved when it is true. I don't follow how you are reading the doc in that way. Is it because of the first paragraph? That's essentially a "background" section.

Reply:
How about "fail-fast" instead of "non-wait-for-ready" ?

Reply:
fail-fast is fine.